### PR TITLE
feat: pass the storage pointer

### PR DESCRIPTION
**Pull request by Schlagonia**
## Description

Pass the storage pointer to all internal functions to avoid reloading it multiple times a call.

Fixes # (issue)

## Checklist

- [x] I have run solidity linting
- [x] I have run the tests on my machine
- [x] I have followed commitlint guidelines
- [x] I have rebased my changes to the latest version of the main branch
- [x] I have updated the SPECIFICATION.md for any relevant changes

### CODE REVIEW

File: src/TokenizedStrategy.sol [hunk 1]
```diff
@@ -201,7 +201,7 @@ contract TokenizedStrategy {
      *
      * Loading the corresponding storage slot for the struct does not
      * load any of the contents of the struct into memory. So the size
-     * has no effect on gas usage.
+     * wont increase gas usage.
```
Comment (fp-crypto) [inline src/TokenizedStrategy.sol:204]:
```suggestion
     * won't increase gas usage.
```

File: src/TokenizedStrategy.sol [hunk 2]
```diff
@@ -498,15 +502,16 @@ contract TokenizedStrategy {
         uint256 assets,
         address receiver
     ) external nonReentrant returns (uint256 shares) {
+        StrategyData storage S = _strategyStorage();
```
Comment (fp-crypto) [inline src/TokenizedStrategy.sol:505]: So TLDR, external/public functions load the point, internal/private functions receive the pointer?

File: src/TokenizedStrategy.sol [hunk 3]
```diff
@@ -642,14 +677,8 @@ contract TokenizedStrategy {
...
-    function convertToAssets(uint256 shares) public view returns (uint256) {
-        // Saves an extra SLOAD if totalSupply() is non-zero.
-        uint256 supply = totalSupply();
-
-        return
-            supply == 0
-                ? shares
-                : shares.mulDiv(totalAssets(), supply, Math.Rounding.Down);
+    function convertToAssets(uint256 shares) external view returns (uint256) {
+        return _convertToAssets(_strategyStorage(), shares);
```
Comment (fp-crypto) [inline src/TokenizedStrategy.sol:681]: What is the advantage of separating the public and private methods like this? Code paths that have the storage pointer loaded can call the internal/private versions?

**Review by fp-crypto — COMMENTED** [review]
Code looks fine, but my eyes could fail me. Do you feel confident the compiler+test suite would have caught any code paths you might have missed?

File: src/TokenizedStrategy.sol (hunk 2 above)
Comment (Schlagonia) [reply src/TokenizedStrategy.sol:505]: yup

**Comment by Schlagonia** [comment]
> Code looks fine, but my eyes could fail me. Do you feel confident the compiler+test suite would have caught any code paths you might have missed?

Yes I would have to think so. 

Added some additional asserts in the tests so that we now have 100% coverage across the TokenizedStrategy lines and functions

File: src/TokenizedStrategy.sol (hunk 3 above)
Comment (Schlagonia) [reply src/TokenizedStrategy.sol:681]: Yes, _convertToAssets is used by multiple other functions within the contract. So any other function will call the internal one

**Review by fp-crypto — APPROVED** [review]

**Pull request merged**